File: frontend/src/types/maintenance.ts
export interface MaintenanceTask {
  site_id: string;
  task_type: string;
  description: string;
  scheduled_at: string;
  priority: string;
}

export interface MaintenanceRequest {
  site_ids: string[];
}

export interface MaintenanceResponse {
  tasks: MaintenanceTask[];
  total: number;
}

File: backend/app/schemas/maintenance.py
from datetime import datetime
from pydantic import BaseModel


class MaintenanceTask(BaseModel):
    site_id: str
    task_type: str
    description: str
    scheduled_at: datetime
    priority: str


class MaintenanceRequest(BaseModel):
    site_ids: list[str]


class MaintenanceResponse(BaseModel):
    tasks: list[MaintenanceTask]
    total: int
